internal/file/file_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file_test
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"testing"
+
+	digest "github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras/internal/file"
+)
+
+const blobMediaType = "application/mock-octet-stream"
+
+func TestFile_PrepareContent(t *testing.T) {
+	// generate test content
+	tempDir := t.TempDir()
+	content := []byte("hello world!")
+	fileName := "test.txt"
+	path := filepath.Join(tempDir, fileName)
+	if err := os.WriteFile(path, content, 0444); err != nil {
+		t.Fatal("error calling WriteFile(), error =", err)
+	}
+
+	want := ocispec.Descriptor{
+		MediaType: blobMediaType,
+		Digest:    digest.FromBytes(content),
+		Size:      int64(len(content)),
+	}
+
+	// test PrepareContent
+	got, rc, err := file.PrepareContent(path, blobMediaType)
+	defer rc.Close()
+	if err != nil {
+		t.Fatal("PrepareContent() error=", err)
+	}
+	_, err = io.ReadAll(rc)
```

Comment:
Returned `[]bytes` should equals to `content`

Reply:
Made requested change